components/psa/services/psa_prot_internal_storage/COMPONENT_PSA_SRV_IMPL/pits_impl.cpp

```diff
@@ -0,0 +1,237 @@
+/* Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cstring>
+#include "KVMap.h"
+#include "KVStore.h"
+#include "TDBStore.h"
+#include "psa_prot_internal_storage.h"
+#include "pits_impl.h"
+#include "mbed_error.h"
+
+#ifdef   __cplusplus
+extern "C"
+{
+#endif
+
+using namespace mbed;
+
+#define PSA_ITS_FILENAME_LEN_BYTES      14          // Maximum length of filename we use for kvstore API.
+                                                    // uid: 6; delimiter: 1; pid: 6; str terminator: 1
+
+
+const uint8_t base64_coding_table[] = {
+    'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H',
+    'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P',
+    'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X',
+    'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f',
+    'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n',
+    'o', 'p', 'q', 'r', 's', 't', 'u', 'v',
+    'w', 'x', 'y', 'z', '0', '1', '2', '3',
+    '4', '5', '6', '7', '8', '9', '+', '-'
+};
+
+
+static KVStore *get_kvstore_instance(void)
+{
+    KVMap &kv_map = KVMap::get_instance();
+
+    return kv_map.get_main_kv_instance("/kv/");//MBED_CONF_STORAGE_DEFAULT_KV); // Danny-TODO: Use configuration name
+}
+
+static void generate_fn(uint8_t *tdb_filename, uint32_t uid, uint32_t pid)
+{
+    // Note: tdb_filename assumed to be of size PSA_ITS_FILENAME_LEN_BYTES and reset with zeros
+
+    uint8_t filename_idx = 0;
+    uint32_t tmp_uid = uid;
+    uint32_t tmp_pid = pid;
+
+    // Iterate on UID; each time convert 6 bits of UID into a character; first iteration must be done
+    do {
+        tdb_filename[filename_idx++] = base64_coding_table[tmp_uid & 0x3F];
+        tmp_uid = tmp_uid >> 6;
+    } while (tmp_uid != 0);
+
+    // Write delimiter
+    tdb_filename[filename_idx++] = '#';
+
+    // Iterate on PID; each time convert 6 bits of PID into a character; first iteration must be done
+    do {
+        tdb_filename[filename_idx++] = base64_coding_table[tmp_pid & 0x3F];
+        tmp_pid = tmp_pid >> 6;
+    } while (tmp_pid != 0);
+}
+
+
+psa_its_status_t psa_its_set_impl(uint32_t pid, uint32_t uid, uint32_t data_length, const void *p_data, psa_its_create_flags_t create_flags)
+{
+    if((create_flags != 0) && (create_flags != PSA_ITS_WRITE_ONCE_FLAG)) {
+        return PSA_ITS_ERROR_FLAGS_NOT_SUPPORTED;
+    }
+
+    KVStore *kvstore = get_kvstore_instance();
```

Comment:
in what cases will we actually receive NULL pointer?
I think NULL pointer means system miss-configuration at build time (all kind of JSON files), so why propagating this type of error code will help system integrator/application developer?
i suggest call panic instead. 
@dreemkiller @davidsaada - what do you suggest?

Reply:
It is a desired behavior. The error condition can ONLY be created during build time configuration.
User cannot cases it by bad input. So there is no need to hide such an error or propagate it.
Calling this function is also a build time decision.
Please replace with calls to `error(...)` function.